modules/storage/cosmos-db/main.bicep

```
@@ -20,65 +16,93 @@ param enableMultipleWriteLocations bool = true
 param enableFreeTier bool = false
 
 @description('The total throughput limit of the Cosmos DB account in measurement of requests units (RUs) per second, -1 indicates no limits on provisioning of throughput.')
 @minValue(-1)
 param totalThroughputLimit int = -1
 
-@description('The array of secondary locations.')
-param secondaryLocations array = []
-
-@allowed([ 'Eventual', 'ConsistentPrefix', 'Session', 'BoundedStaleness', 'Strong' ])
-@description('The default consistency level. Possible values: "Eventual", "ConsistentPrefix", "Session", "BoundedStaleness", "Strong".')
-param defaultConsistencyLevel string = 'Session'
-
-@description('Max stale requests required for "BoundedStaleness" Consistency Level. Valid ranges, Single Region: 10 to 2147483647. Multi Region: 100000 to 2147483647.')
-@minValue(10)
-@maxValue(2147483647)
-param maxStalenessPrefix int = 100000
-
-@description('Max lag time (minutes). Required for BoundedStaleness. Valid ranges, Single Region: 5 to 84600. Multi Region: 300 to 86400.')
-@minValue(5)
-@maxValue(86400)
-param maxIntervalInSeconds int = 300
+@minLength(1)
+@description('''
+The array of secondary locations.
+Each element defines a region of georeplication. The first element in this array is the primary region which is a write region of the Cosmos DB account.
+The order of regions in this list is the order of regions to be used for failover.
+Parameters:
+- Name: name
+  Description: The name of the Azure region.
+- Name: isZoneRedundant
+  Description: Flag to indicate whether or not this region is an AvailabilityZone region
+''')
+param locations {
+  name: string
+  isZoneRedundant: bool?
+}[]
 
 @description('MongoDB server version. Required for mongodb API type Cosmos DB account')
 @allowed([ '3.2', '3.6', '4.0', '4.2' ])
-param serverVersion string = '4.2'
-
-@description('List of CORS rules.')
-param cors array = []
+param MongoDBServerVersion string = '4.2'
+
+@description('''
+List of CORS rules. Each CORS rule allows or denies requests from a set of origins to a Cosmos DB account or a database.
+parameters:
+- Name: allowedOrigins
+  Description: The origin domains that are permitted to make a request against the service via CORS.
+- Name: allowedMethods
+  Description: The methods (HTTP request verbs) that the origin domain may use for a CORS request. (comma separated)
+- Name: allowedHeaders
+  Description: The response headers that should be sent back to the client for CORS requests. (comma separated)
+- Name: exposedHeaders
+  Description: The response headers that should be exposed to the client for CORS requests. (comma separated)
+- Name: maxAgeInSeconds
+  Description: The maximum amount time that a browser should cache the preflight OPTIONS request.
+''')
+param cors {
+  allowedOrigins: string
+  allowedMethods: string?
+  allowedHeaders: string?
+  exposedHeaders: string?
+  maxAgeInSeconds: int?
+}[] = []
```

```suggestion
param cors corsType = []

type corsType = {
  @description('The origin domains that are permitted to make a request against the service via CORS.')
  allowedOrigins: string
  allowedMethods: string?
  allowedHeaders: string?
  exposedHeaders: string?
  maxAgeInSeconds: int?
}[]
```

this looks a good spot to try using a custom type as well. You can add the descriptions directly to the types, and reduce the content in the param description.